src/metatensor/models/experimental/pet/train.py

```diff
@@ -0,0 +1,142 @@
+    if len(validation_datasets) != 1:
+        raise ValueError("PET only supports a single validation dataset")
+
+    if device_str == "gpu":
+        device_str = "cuda"
+
+    if continue_from is not None:
+        hypers["FITTING_SCHEME"]["MODEL_TO_START_WITH"] = continue_from
+
+    train_dataset = train_datasets[0]
+    validation_dataset = validation_datasets[0]
+
+    # dummy dataloaders due to https://github.com/lab-cosmo/metatensor/issues/521
+    train_dataloader = DataLoader(
+        train_dataset,
+        batch_size=1,
+        shuffle=False,
+        collate_fn=collate_fn,
+    )
+    validation_dataloader = DataLoader(
+        validation_dataset,
+        batch_size=1,
+        shuffle=False,
+        collate_fn=collate_fn,
+    )
+
+    # only energies or energies and forces?
+    do_forces = next(iter(train_dataset))[1].block().has_gradient("positions")
+    all_species = requested_capabilities.species
+    if not do_forces:
+        hypers["MLIP_SETTINGS"]["USE_FORCES"] = False
+
+    ase_train_dataset = []
+    for (system,), targets in train_dataloader:
+        ase_atoms = system_to_ase(system)
+        ase_atoms.info["energy"] = float(
+            targets[target_name].block().values.squeeze(-1).detach().cpu().numpy()
+        )
+        if do_forces:
+            ase_atoms.arrays["forces"] = (
+                -targets[target_name]
+                .block()
+                .gradient("positions")
+                .values.squeeze(-1)
+                .detach()
+                .cpu()
+                .numpy()
+            )
+        ase_train_dataset.append(ase_atoms)
+
+    ase_validation_dataset = []
+    for (system,), targets in validation_dataloader:
+        ase_atoms = system_to_ase(system)
+        ase_atoms.info["energy"] = float(
+            targets[target_name].block().values.squeeze(-1).detach().cpu().numpy()
+        )
+        if do_forces:
+            ase_atoms.arrays["forces"] = (
+                -targets[target_name]
+                .block()
+                .gradient("positions")
+                .values.squeeze(-1)
+                .detach()
+                .cpu()
+                .numpy()
+            )
+        ase_validation_dataset.append(ase_atoms)
+
+    fit_pet(
+        ase_train_dataset, ase_validation_dataset, hypers, "pet", device_str, output_dir
+    )
+
+    if do_forces:
+        load_path = Path(output_dir) / "pet" / "best_val_rmse_forces_model_state_dict"
+    else:
+        load_path = Path(output_dir) / "pet" / "best_val_rmse_energies_model_state_dict"
+
+    state_dict = torch.load(load_path)
+
+    ARCHITECTURAL_HYPERS = Hypers(hypers["ARCHITECTURAL_HYPERS"])
+    ARCHITECTURAL_HYPERS.D_OUTPUT = 1  # energy is a single scalar
+    ARCHITECTURAL_HYPERS.TARGET_TYPE = "structural"  # energy is structural property
+    ARCHITECTURAL_HYPERS.TARGET_AGGREGATION = (
+        "sum"  # energy is a sum of atomic energies
+    )
```

Comment:
These coefs are already hardcoded in the `model.py`

Reply:
Hmmm fair, I need to discuss a few things with @serfg but then I agree that some hypers are not meant to be changed (for now)